src/java/org/apache/cassandra/db/marshal/ListType.java

```diff
@@ -127,7 +127,7 @@ public ListSerializer<T> getSerializer()
     public AbstractType<?> freeze()
     {
         if (isMultiCell)
-            return getInstance(this.elements, false);
+            return getInstance(this.elements.freeze(), false);
```

Comment:
need to think about this more... if the parent is frozen then it really doesn't matter if the children are or not...  but it does for equality and for the type tests...